ext/node/crypto/mod.rs

```diff
@@ -153,3 +154,33 @@ pub fn op_node_public_encrypt(
     _ => Err(type_error("Unknown padding")),
   }
 }
+
+#[op(fast)]
+pub fn op_node_create_cipheriv(
+  state: &mut OpState,
+  algorithm: &str,
+  key: &[u8],
+  iv: &[u8],
+) -> u32 {
+  state.resource_table.add(
+    match cipher::CipherContext::new(algorithm, key, iv) {
+      Ok(context) => context,
+      Err(_) => return 0,
+    },
+  )
+}
+
+#[op(fast)]
+pub fn op_node_cipheriv_encrypt(
+  state: &mut OpState,
+  rid: u32,
+  input: &[u8],
+  output: &mut [u8],
+) -> bool {
+  let context = match state.resource_table.get::<cipher::CipherContext>(rid) {
+    Ok(context) => context,
+    Err(_) => return false,
+  };
```

Comment:
This should just consume the resource i.e `resource_table.take`. We don't need to call `core.close` from JS

Reply:
Added `op_node_cipheriv_final` which is called at the end of encryption and consumes the resource. (`op_node_cipheriv_encrypt` was used both for the last and middle encryptions, but that was probably confusing)